c/polyjuice.h

```diff
@@ -1387,6 +1457,63 @@ int run_polyjuice() {
     return ret;
   }
 
+  /**
+   * We seperate two branches: 
+   * - transfer native token to EOA account
+   * - the rest
+   *
+   * The part of transferring native tokens to EOA account is isolated. It will
+   * not enter into EVM. 
+   * Recognizing EOA transferring if conditions are satisfied below:
+   * - to_id is g_creator_account_id
+   * - only accept call_kind == EVMC_CALL
+   * - g_eoa_transfer_flag is true
+   * - g_eoa_transfer_to_address is not zero address
+   * The `g_eoa_transfer_to_address` which is the true `to_address` that is
+   * going to transfer to, and must not be a contract address.
+   * 
+   * Regarding transfer to contract account, a normal polyjuice transaction
+   * which `to_id` is the contract account should be expected.
+   *
+   **/
+  if (g_creator_account_id == context.transaction_context.to_id && 
+          msg.kind == EVMC_CALL &&
```

Comment:
> We should also check the call_kind, we can only accept transfer with CALL call_kink.

@jjyr fixed

Reply:
Can we also check the call_kind in the parsing phase? Invalid transactions should fail fast.

Reply:
Done!